frontend/src/utils/lessonUtils.ts

```diff
@@ -1,36 +1,100 @@
 import { v4 as uuid } from "uuid";
 
 import {
+  ContentBlockDTO,
   LessonRequest,
   LessonResponse,
 } from "../APIClients/types/LessonClientTypes";
 import { LessonType } from "../types/ModuleEditorTypes";
-import { ContentBlockState, ContentTypeEnum } from "../types/ContentBlockTypes";
+import {
+  ColumnBlockState,
+  ContentBlockState,
+  ContentTypeEnum,
+} from "../types/ContentBlockTypes";
 
-export const formatLessonRequest = (lesson: LessonType): LessonRequest => {
-  // Remove IDs from content blocks, switch types to the titles
-  const content = lesson.content.map((block) => ({
+function isRecord(value: unknown): value is Record<string, unknown> {
+  return typeof value === "object" && value !== null;
+}
+
+const isBlockDTO = (block: unknown): block is ContentBlockDTO => {
+  if (!isRecord(block)) return false;
+  if (!(typeof block.type === "string")) return false;
+  if (!isRecord(block.content)) return false;
+  return true;
+};
+
+const isColumnState = (block: ContentBlockState): block is ColumnBlockState => {
+  return block.type.id === ContentTypeEnum.COLUMN.id;
+};
+
+const isContentBlock = (block: unknown): block is ContentBlockState => {
+  if (!isRecord(block)) return false;
+  if (!isRecord(block.type)) return false;
+  if (!isRecord(block.content)) return false;
+  if (!(typeof block.type.id === "string")) return false;
+  if (!(typeof block.id === "string")) return false;
+  return true;
+  // TODO: Can we create an object from the typescript definitions ?
+  // const testObj = createContent(block.type.id);
```

Comment:
Good idea! I think we could, but it would have to be the other way 'round, i.e., creating the Typescript definitions from the objects. Definitely a nice-to-have though and probably can juts leave this TODO for now.